pkg/controller/datadogagent/datadogagent_controller_test.go

```diff
@@ -2103,6 +2165,34 @@ func hasAllClusterLevelRbacResources(policyRules []rbacv1.PolicyRule) bool {
 	return len(clusterLevelResources) == 0
 }
 
+func hasWpaRbacs(policyRules []rbacv1.PolicyRule) bool {
+	resourceFound := false
+	groupFound := false
+	verbsFound := false
+	requiredVerbs := []string{
+		datadoghqv1alpha1.ListVerb,
+		datadoghqv1alpha1.WatchVerb,
+		datadoghqv1alpha1.GetVerb,
+	}
+	for _, policyRule := range policyRules {
+		for _, resource := range policyRule.Resources {
+			if resource == "watermarkpodautoscalers" {
+				resourceFound = true
+			}
+		}
+		for _, group := range policyRule.APIGroups {
+			if group == "datadoghq.com" {
+				groupFound = true
+			}
+		}
+		if reflect.DeepEqual(policyRule.Verbs, requiredVerbs) {
+			verbsFound = true
+		}
+	}
+
+	return resourceFound && groupFound && verbsFound
+}
```

Comment:
It looks like this function will return `true` even for something like:
```yaml
- apiGroups:
  - "datadoghq.com"
  resources:
  - "datadogagent"
  verbs:
  - "create"
- apiGroups:
  - "something.unrelated.to.datadog"
  resources:
  - "watermarkpodautoscalers"
  verbs:
  - "update"
- apiGroups:
  - ""
  resources:
  - "pods"
  verbs:
  - "list"
  - "get"
  - "watch"
```
whereas we expect it to return `true` only when all the required elements are in the same list item:
```yaml
- apiGroups:
  - "datadoghq.com"
  resources:
  - "watermarkpodautoscalers"
  verbs:
  - "list"
  - "get"
  - "watch"
```

```suggestion
func hasWpaRbacs(policyRules []rbacv1.PolicyRule) bool {
	requiredVerbs := []string{
		datadoghqv1alpha1.ListVerb,
		datadoghqv1alpha1.WatchVerb,
		datadoghqv1alpha1.GetVerb,
	}

	for _, policyRule := range policyRules {
		resourceFound := false
		groupFound := false
		verbsFound := false

		for _, resource := range policyRule.Resources {
			if resource == "watermarkpodautoscalers" {
				resourceFound = true
				break
			}
		}
		for _, group := range policyRule.APIGroups {
			if group == "datadoghq.com" {
				groupFound = true
				break
			}
		}
		if reflect.DeepEqual(policyRule.Verbs, requiredVerbs) { // Test to be updated
			verbsFound = true
			break
		}
		
		if resourceFound && groupFound && verbsFound {
			return true
		}
	}

	return false
}
```